packages/perspective-viewer-d3fc/src/js/legend/styling/enableDragging.js

```diff
@@ -0,0 +1,48 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2017, the Perspective Authors.
+ *
+ * This file is part of the Perspective library, distributed under the terms of
+ * the Apache License 2.0.  The full license can be found in the LICENSE file.
+ *
+ */
+
+export function enableDragging(element) {
+    const node = element.node();
+    node.style.cursor = "move";
+
+    let xPositionDiff = 0,
+        yPositionDiff = 0,
+        xPositionLast = 0,
+        yPositionLast = 0;
+
+    node.onmousedown = dragMouseDown;
+
+    function dragMouseDown(e) {
+        e = e || window.event;
+        e.preventDefault();
+        xPositionLast = e.clientX;
+        yPositionLast = e.clientY;
+        document.onmouseup = releaseNode;
+        document.onmousemove = dragNode;
+    }
+
+    function dragNode(e) {
+        e = e || window.event;
+        e.preventDefault();
+        // calculate the new cursor position:
+        xPositionDiff = xPositionLast - e.clientX;
+        yPositionDiff = yPositionLast - e.clientY;
+        xPositionLast = e.clientX;
+        yPositionLast = e.clientY;
+        // set the node's new position:
+        node.style.top = node.offsetTop - yPositionDiff + "px";
+        node.style.left = node.offsetLeft - xPositionDiff + "px";
```

Comment:
There doesn't seem to be any protection against dragging it completely off the screen (making it un-recoverable).

Reply:
Interesting point. Also discovered with Ronan, that the behaviour regarding at the extreme edges can become a bit strange, with the text overflowing the box.

I'll investigate this stuff. There is a d3 dragging behaviour so I'll have a look at all that first.

Reply:
No dragging off the screen anymore.